include/ProcessTools.h

```diff
@@ -45,25 +45,93 @@
+    /// to be the path to the executable to be launched. If that executable has a space in it's path, use
+    /// the two parameter version of this function instead.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns the ExitCode and Cout/Cerr output of the command that was run.
+    CommandResult MEZZ_LIB RunCommand(const StringView Command);
+
+    /// @brief Launches a different process on the system and return only it's console output as a String.
+    /// @note ExecutablePath cannot be empty on Posix systems or this function will fail.
+    /// @remarks This function ignores the exit code of the command. If you want/need the exit code, use
+    /// @ref RunCommand instead.
+    /// @param ExecutablePath The path to the executable to be invoked.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns a String containing the output of the specified command.
+    [[nodiscard]]
+    String MEZZ_LIB GetCommandOutput(const StringView ExecutablePath, const StringView Command);
+    /// @brief Launches a different process on the system and return only it's console output as a String.
+    /// @remarks This function ignores the exit code of the command. If you want/need the exit code, use
+    /// @ref RunCommand instead. @n@n
+    /// This function will interpret all of the text up until the first delimiter (space or tab) to be the
+    /// path to the executable to be launched. If that executable has a space in it's path, use the two
+    /// parameter version of this function instead.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns a String containing the output of the specified command.
+    [[nodiscard]]
+    String MEZZ_LIB GetCommandOutput(const StringView Command);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Output to File
+
+    /// @brief Run a process and direct its console output to a file.
+    /// @note ExecutablePath cannot be empty on Posix systems or this function will fail.
+    /// @remarks This function is meant to be used in conjunction with GetFileContents to view the output.
+    /// @details This launches a process, redirects its output to a file and returns the command's exit code.
+    /// This is a simple wrapper around std::system, concatenating output redirection to the command. Because of
+    /// this "|", "<" and ">" are not allowed in commands passed to this.
+    /// @param ExecutablePath The path to the executable to be invoked.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @param OutputFileName The file to put all the console output of the command.
+    /// @return Returns the exit code of the command.
+    Integer MEZZ_LIB OutputCommandToFile(const StringView ExecutablePath,
```

Comment:
I though when I made these they were just an implementation detail of GetCommandOutput. Can these be removed... May we should keep them. Is this that common of a use case around running external processes?

I haven't read it all yet, I am eager to see the implementation details.

Reply:
I don't know exactly if they are a common use case at all.  But we could remove it.  It's a little bit awkward for it to exist at all if I'm being honest.  I just didn't want to regress in something we used to have.

Reply:
This used to exist to work around the limits of std::system. It only returned an error code, and I wanted the text. If this is better now then isn't needed now.